src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,134 @@
+        actual.setAge(null);
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(actual));
+    }
+
+    @Test
+    void register_lessMinAge_notOk() {
+        User actual = testUser;
+        actual.setAge(8);
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(actual));
+        actual.setAge(17);
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(actual));
+    }
+
+    @Test
+    void registration_negativeAge_notOk() {
+        User actual = testUser;
+        actual.setAge(-18);
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(actual));
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        User actual = testUser;
+        actual.setLogin(null);
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(actual));
+    }
+
+    @Test
+    void register_shortLogin_notOk() {
+        User actual = testUser;
+        actual.setLogin("log");
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(actual));
+        actual.setLogin("login");
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(actual));
+    }
+
+    @Test
+    void register_loginIsTaken_notOk() {
+        storageDao.add(testUser);
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(testUser));
+    }
+    
+    @Test
+    void register_nullPassword_notOk() {
+        User actual = testUser;
+        actual.setPassword(null);
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(actual));
+    }
+
+    @Test
+    void register_shortPassword_notOk() {
+        User actual = testUser;
+        actual.setPassword("pas");
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(actual));
+        actual.setPassword("passw");
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(actual));
+    }
+    
+    @Test
+    void register_loginIsTakenAfterRegistration_ok() {
+        registrationService.register(testUser);
+        String actual = storageDao.get(testUser.getLogin()).getLogin();
+        String expected = testUser.getLogin();
+        assertEquals(expected,actual);
+    }
+
+    @Test
+    void register_minLengthLoginPassword_ok() {
+        User actual = testUser;
+        actual.setAge(18);
+        actual.setLogin("logins");
+        actual.setPassword("passwo");
+        assertDoesNotThrow(() -> registrationService.register(actual));
+    }
+
+    @Test
+    void register_validDataUser_ok() {
+        User actual = testUser;
+        assertDoesNotThrow(() -> registrationService.register(actual));
+    }
```

Comment:
![image](https://github.com/mate-academy/jv-registration-validation/assets/49767049/25615610-d870-41cf-a0a5-0e94f4d171b3)
it's not enough to check nothing is thrown. Also, better put ok test first